File: js/test/test_ipv6_loopback.ts
describe("ipv6_loopback", () => {
    class IPv6LoopbackTest {
        ip: IPAddress,
        s: string,
        n: string,
        string: string,
        one: Crunchy
    }
    function setup() : IPv6LoopbackTest {
        return IPv6LoopbackTest {
            ip: ipv6_loopback::new(),
            s: "::1",
            n: "::1/128",
            string: "0000:0000:0000:0000:0000:0000:0000:0001/128",
            one: BigUint::one(),
        };
    }

    it("test_attributes", () => {
        let s = setup();
        assert_eq!(128, s.ip.prefix.num);
        assert_eq!(true, s.ip.is_loopback());
        assert_eq!(s.s, s.ip.to_s());
        assert_eq!(s.n, s.ip.to_string());
        assert_eq!(s.string, s.ip.to_string_uncompressed());
        assert_eq!(s.one, s.ip.host_address);
    });
    it("test_method_ipv6", () => {
        assert_eq!(true, setup().ip.is_ipv6());
    });
});

File: cpp/test/test_ipv6_loopback.cpp
#include "chai.hpp"
#include "mocha.hpp"

#include "../src/ipaddress.hpp"
#include "../src/prefix128.hpp"
#include "../src/ipv6_loopback.hpp"
#include "../src/crunchy.hpp"

typedef struct S_IPv6LoopbackTest {
  IPAddress ip;
  const char * s;
  const char * n;
  const char *string;
  Crunchy one;
} IPv6LoopbackTest;

IPv6LoopbackTest setup() {
  IPv6LoopbackTest ret = {
    Ipv6Loopback::create(),
    "::1",
    "::1/128",
    "0000:0000:0000:0000:0000:0000:0000:0001/128",
    Crunchy::one()
  };
  return ret;
}

int main() {

  describe("ipv6_loopback", []() {
      it("test_attributes", []() {
          auto s = setup();
          Chai::assert.equal(128, s.ip.prefix.num);
          Chai::assert.equal(true, s.ip.is_loopback());
          Chai::assert.equal(s.s, s.ip.to_s());
          Chai::assert.equal(s.n, s.ip.to_string());
          Chai::assert.equal(s.string, s.ip.to_string_uncompressed());
          Chai::assert.equal(s.one.toString(), s.ip.host_address.toString());
          });
      it("test_method_ipv6", []() {
          Chai::assert.equal(true, setup().ip.is_ipv6());
          });
      });
  return exit();
}
